addons/block_code/ui/util.gd

```diff
@@ -12,3 +12,38 @@ static func node_is_part_of_edited_scene(node: Node) -> bool:
 
 	var edited_scene_parent := tree.edited_scene_root.get_parent()
 	return edited_scene_parent and edited_scene_parent.is_ancestor_of(node)
+
+
+## Get the path from [param reference] to [param node] within a scene.
+##
+## Returns the path from [param reference] to [param node] without referencing
+## parent nodes. If [param node] is [param reference] or a child of it in the
+## scene tree, a relative path is returned. If [param node] is an ancestor of
+## [param reference] in the scene tree, an absolute path using [param
+## path_root] is returned.
+## [br]
+## Both [param node] and [param reference] must be ancestors of [param
+## path_root]. If [param path_root] is [constant null]
+## [method EditorInterface.get_edited_scene_root().get_parent] is used.
+static func node_scene_path(node: Node, reference: Node, path_root: Node = null) -> NodePath:
```

Comment:
This is great. I wonder if you had to read the editor source for reference or implemented it from scratch.

Reply:
It was a lot of trial and error and printing out paths until I felt reasonably comfortable that I knew the pattern. I also had to read the SceneTree, NodePath and Node documentation many times. In particular, I didn't understand that `get_edited_scene_root()` or `SceneTree.edited_scene_root` _don't_ correspond to `/root`. `/root` corresponds to the parent of the scene root node. That's `SceneTree.root`, but in the editor that's the entire editor window. Hence why it does `get_edited_scene_root().get_parent()` since you're deep in the editor hierarchy in our cases.

I'm sure there are some corner cases that aren't covered, but I think it will fit our needs.

Reply:
Yeah this is certainly something to improve upstream. Why dragging a node from the current scene tree would give you a path relative to the editor root?

Reply:
Because the current scene _is_ the whole editor. The path you're given is an absolute path that's entirely valid while the editor is running because that's where it is in the (singleton) `SceneTree`.

I should have put this in a comment, but this is a special case for us. We're trying to create an absolute path for when `edited_scene_root` later becomes `current_scene`. The absolute '/root` path returned from here is bogus while you're in the editor. It only becomes valid when the scene is actually running. That's what we need here because we're generating a script that won't actually run until the edited scene becomes the current scene and is at the root of the scene tree.

If you just want an absolute path to the node in whatever is the current scene, you can just use `Node.get_path()`. Originally I wrote this to always use relative paths as those work no matter where the scene is in the scene tree. That's also why you use relative paths such as `$Foo/Bar` (equivalent to `get_node("Foo/Bar")`) in GDScript - they work regardless of where the node using the script is in the scene tree. However, relative paths with `../../SomeNode` seemed too hard to grasp, which is why I came up with the  "absolute path that will become valid at runtime" thing.

So, if you wanted to upstream this, it would have to be a separate method like `get_edited_path(relative: bool = false)` that returns a path treating the edited scene as if it was the current scene.